src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.CSharp.Diagnostics.SimplifyTypeNames;
+using Microsoft.CodeAnalysis.CSharp.TypeStyle;
+using Microsoft.CodeAnalysis.CSharp.UseExpressionBody;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics.SimplifyTypeNames;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Formatting
+{
+    [UseExportProvider]
+    public class CodeCleanupTests
```

Comment:
>CodeCleanupTests [](start = 17, length = 16)

Synced up with @carhu@microsoft.com offline and she plans to extend the tests, but probably not as part of this PR due to time constraints. I think we should at least have the following tests:
1) Simple unit tests, one per option, that verify code cleanup works fine with only single option enabled.
2) Unit tests with multiple options enabled, but the reported diagnostics are not conflicting, hence can all be fixed.
3) Unit tests with multiple options enabled, where reported diagnostics have conflicting spans and hence not all can be fixed. Is this possible for our currently chosen diagnostics/fixers?
4) Verify options persistence after changing the workspace option.

Reply:
thanks, copied into https://github.com/dotnet/roslyn/issues/27497